tests/lean/run/smt2_P_implies_Q_fail.lean

```diff
@@ -0,0 +1,5 @@
+example (P Q : Prop) : P -> Q :=
+begin
+    intros,
+    z3
```

Comment:
This introduces a hard build dependency on Z3 in order to run the tests.  I think we should move the Z3 tests into a separate directory and only run them if we can find the `z3` executable.